### ISSUE TITLE
Add structured Logger interface 

### ISSUE THREAD

**Issue by basvanbeek** [openzipkin/zipkin-go#37]
We should provide a standard structured logger interface which consumers can inject with an implementation. `zipkin-go-opentracing` uses an interface compatible with the `Go kit` logger interface which seems the best candidate.

```
type Logger interface {
	Log(keyvals ...interface{}) error
}
```

This will allow people to easily fetch the K/V pairs coming from errors and exceptions within the library and merge them with their application logging infrastructure.

For people not liking structured logging we should provide an adapter for the standard Go log.

**Comment by adriancole** [comment]
Ahh so this is so that the logging internal to the library is abstracted,
right? Ex this is only used by tracing code.

On 23 Dec 2017 1:26 am, "Bas van Beek" <notifications@github.com> wrote:

We should provide a standard structured logger interface which consumers
can inject with an implementation. zipkin-go-opentracing uses an interface
compatible with the Go kit logger interface which seems the best candidate.

type Logger interface {
	Log(keyvals ...interface{}) error
}

This will allow people to easily fetch the K/V pairs coming from errors and
exceptions within the library and merge them with their application logging
infrastructure.

For people not liking structured logging we should provide an adapter for
the standard Go log.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/openzipkin/zipkin-go/issues/37>, or mute the thread

**Comment by basvanbeek** [comment]
Yes that is correct. It allows the user of the library decide how to have errors be reported.

**Comment by kujenga** [comment]
I took a pass at implementing a more complete version of this to use interfaces throughout the package: https://github.com/openzipkin/zipkin-go/pull/188

**Comment by basovnik** [comment]
I would appreciate this feature as well. I would like to integrate [zap](https://github.com/uber-go/zap) logger using an interface.
